plugins/plugins.go

```diff
@@ -25,8 +32,18 @@ type Plugin interface {
 type Manager struct {
 	Labels   map[string]string
 	Store    storage.Store
+	Compiler *ast.Compiler
 	services map[string]rest.Client
 	plugins  []Plugin
+	mtx      sync.RWMutex
+}
+
+// CompilerTriggerConfig contains the trigger registration configuration for
+// compiler changes
+type CompilerTriggerConfig struct {
```

Comment:
I feel like having a separate config struct (like we do in the `storage` package) is overkill here. We have it in storage because we could potentially register triggers on for commit, or abort, or on specific paths. In this case, we only need to notify listeners (like the server) when the compiler changes. A function like `func(*ast.Compiler, storage.Transaction)` should be sufficient.

Reply:
Removed the config